packages/block/src/block.ts

```diff
@@ -299,11 +316,76 @@ export class Block {
     }
   }
 
-  private _validateUncleHeader(uncleHeader: BlockHeader, blockchain: Blockchain) {
-    // TODO: Validate that the uncle header hasn't been included in the blockchain yet.
-    // This is not possible in ethereumjs-blockchain since this PR was merged:
-    // https://github.com/ethereumjs/ethereumjs-blockchain/pull/47
-    const height = this.header.number
-    return uncleHeader.validate(blockchain, height)
+  private async _validateUncleHeaders(uncleHeaders: BlockHeader[], blockchain: Blockchain) {
```

Comment:
Is there any good reason why these validation methods needing a block are here in the `Block` library at all and not in `Blockchain`? 🤔

Nothing for v5 any more but depending on discussion outcome we might want to refactor and put this into `Blockchain` as a refactoring target for v6.

Reply:
Note: have updated the comment, please read on GitHub.

Reply:
I would tend to put the comments below up to the doc-visible comments section, but decide for yourself.

Reply:
The idea of making `blockchain` validate `block` seems like one we should explore; this practically means that the validation logic goes into `blockchain` and the data is verified in `header.ts` and `block.ts` (uncle hash). It would make sense to do that.

Reply:
Opened an issue for this discussion; #947